TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/mediation/smart/SmartWebViewFragment.kt

```diff
@@ -0,0 +1,123 @@
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_inread_webview, container, false)
+    }
+
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
+        TeadsHelper.initialize()
+
+        adView = SASBannerView(requireContext())
+
+        webviewHelperSynch = SyncAdWebView(requireContext(), webview, this@SmartWebViewFragment, "#teads-placement-slot")
+
+        SASConfiguration.getSharedInstance().configure(requireContext(), siteID.toInt())
+        SASConfiguration.getSharedInstance().isLoggingEnabled = true
+
+        webviewHelperSynch.registerAdView(adView)
+
+        mListener = object : TeadsAdapterListener {
+            override fun adOpportunityTrackerView(trackerView: AdOpportunityTrackerView) {
+                webviewHelperSynch.registerTrackerView(trackerView)
+            }
+
+            override fun onRatioUpdated(adRatio: AdRatio) {
+                val params: ViewGroup.LayoutParams = adView.layoutParams
+
+                // Here the width of parent is MATCH_PARENT
+                params.height = adRatio.calculateHeight(adView.measuredWidth)
+
+                /* You need to call updateSlot method from the helper
+                Some creative can resize by itself, to handle it we have to notify the webview helper
+                this ratio doesn't contains the footer and the header
+                To manage this behavior, a work around is to substract 0.2 to the media ratio
+                 */
+                webviewHelperSynch.updateSlot(adRatio.getAdSlotRatio(adView.measuredWidth))
+
+                adView.layoutParams = params
+            }
+        }
+
+        if ((activity as MainActivity).isWebViewDarkTheme
+                && WebViewFeature.isFeatureSupported(WebViewFeature.FORCE_DARK)) {
+            WebSettingsCompat.setForceDark(webview.settings, WebSettingsCompat.FORCE_DARK_ON)
+        }
+        webview.settings.javaScriptEnabled = true
+        webview.webViewClient = CustomInReadWebviewClient(webviewHelperSynch, getTitle())
+        webview.loadUrl(this.webViewUrl)
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        webviewHelperSynch.onConfigurationChanged()
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////
+     * WebView helper listener
+     *//////////////////////////////////////////////////////////////////////////////////////////////////
+
+    override fun onHelperReady(adContainer: ViewGroup) {
+        val key = TeadsHelper.attachListener(mListener)
+
+        val extras = TeadsMediationSettings.Builder()
+                // Needed by european regulation
+                // See https://mobile.teads.tv/sdk/documentation/android/gdpr-consent
+                .userConsent("1", "0001", TCFVersion.V1, 12)
```

Comment:
I see we are using user consent TCFVersion.V1 across to sample app, even it being not supported anymore